shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/show/executor/ExportSchemaConfigurationExecutor.java

```diff
@@ -0,0 +1,348 @@
+import org.apache.shardingsphere.sharding.api.config.ShardingRuleConfiguration;
+import org.apache.shardingsphere.sharding.api.config.rule.ShardingAutoTableRuleConfiguration;
+import org.apache.shardingsphere.sharding.api.config.rule.ShardingTableRuleConfiguration;
+import org.apache.shardingsphere.sharding.api.config.strategy.keygen.KeyGenerateStrategyConfiguration;
+import org.apache.shardingsphere.sharding.api.config.strategy.sharding.ComplexShardingStrategyConfiguration;
+import org.apache.shardingsphere.sharding.api.config.strategy.sharding.HintShardingStrategyConfiguration;
+import org.apache.shardingsphere.sharding.api.config.strategy.sharding.NoneShardingStrategyConfiguration;
+import org.apache.shardingsphere.sharding.api.config.strategy.sharding.ShardingStrategyConfiguration;
+import org.apache.shardingsphere.sharding.api.config.strategy.sharding.StandardShardingStrategyConfiguration;
+import org.apache.shardingsphere.sharding.merge.dal.common.MultipleLocalDataMergedResult;
+
+import javax.sql.DataSource;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.sql.Types;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Export schema configuration executor.
+ */
+@RequiredArgsConstructor
+public final class ExportSchemaConfigurationExecutor extends AbstractShowExecutor {
+    
+    private static final String CONFIG = "config";
+    
+    private static final String COLON = ":";
+    
+    private static final String SPACE = " ";
+    
+    private static final String NEWLINE = "\n";
+    
+    private static final String COLON_SPACE = COLON + SPACE;
+    
+    private static final String COLON_NEWLINE = COLON + NEWLINE;
+    
+    private static final String INDENT = SPACE + SPACE;
+    
+    private final ExportSchemaConfigurationStatement sqlStatement;
+    
+    private final ConnectionSession connectionSession;
+    
+    @Override
+    protected List<QueryHeader> createQueryHeaders() {
+        return Arrays.asList(new QueryHeader("", "", CONFIG, CONFIG, Types.VARCHAR, "VARCHAR", 128, 0, false, false, false, false));
+    }
+    
+    @Override
+    protected MergedResult createMergedResult() {
+        String schemaName = getSchemaName();
+        ShardingSphereMetaData metaData = ProxyContext.getInstance().getMetaData(schemaName);
+        StringBuilder result = new StringBuilder();
+        result.append("schemaName").append(COLON_SPACE).append(schemaName).append(NEWLINE);
+        getDataSourcesConfig(metaData, result);
+        Optional<ShardingRuleConfiguration> ruleConfig = metaData.getRuleMetaData().getConfigurations()
+                .stream().filter(each -> each instanceof ShardingRuleConfiguration).map(each -> (ShardingRuleConfiguration) each).findAny();
+        ruleConfig.ifPresent(rule -> getRulesConfig(rule, result));
```

Comment:
Sorry, I only see the code to export the sharding rule, won't other rules be exported?

Reply:
> Sorry, I only see the code to export the sharding rule, won't other rules be exported?

Yes, can I continue to do it in a follow-up pr.

Reply:
Sure.